### PR TITLE
2022-OCT

### LINKED ISSUE #2: 2022-OCT

### PR DESCRIPTION
**Pull request by zhijinggg**

### CODE REVIEW

File: 2022-OCT/answer.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+### Question 1
+
+a) i)
+
+Primary key is a unique identifier for each record in a database table. For instance, in a library management system, we will have a BOOKS table. In the BOOKS table, it contains BookID as a primary key to ensure that no two books can have the same identifier.
+
+a) ii)
+
+For example, a university course enrollment system contains 'Students' table, 'Courses' table and 'Enrollments' table that manages the many to many relationship between students and courses for each semester. Therefore, the 'Enrollments' table will contain three composite keys namely StudentID, CourseID and Semester. The issue with composite key is that other tables that need to reference this 'Enrollments' table must include all parts of the composite key (StudentID, CourseID and Semester) which complicates the foreign key relationships. To improve this design, we can use surrogate key for the 'Enrollments' table namely EnrollmentID. WHen other tables want to reference this 'Enrollments' table, they only need to include the single 'EnrollmentID' column which simplifies foreign key relationships.
```
Comment (YKLoh666) [inline 2022-OCT/answer.md]: A university course enrollment system contains an ENROLLMENT table which uses a composite key to identify unique records, namely StudentID, CourseID and Semester. 

Another table called ATTENDANCE would like to reference this table's composite key. However, the composite key is formed by 3 fields of data, which will take up a substantial amount of space when the ATTENDANCE table is scaling up. 

The solution will be using a surrogate key for the ENROLLMENT table, which is an enrollmentID. With enrollmentID acting as a primary key for the ENROLLMENT table, the ATTENDANCE table can reference it as a foreign key. This reduces the size of the foreign key in the ATTENDANCE table which could save disk space.

File: 2022-OCT/answer.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+b) 
+
+DBMS Functions : 
+- Backup and Recovery management provides backup and data recovery procedures.
+- Data dictionary management defines data elements and their relationships
+- Data storage management stores data and related data entry forms, report definitions etc.
```
Comment (YKLoh666) [inline 2022-OCT/answer.md:33]: 2 marks each, better give one more sentence elaboration

File: 2022-OCT/answer.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+c)
+
+- Data redundancy is when different and conflicting versions of the same data appear in different places.
```
Comment (YKLoh666) [inline 2022-OCT/answer.md]: This is more like the definition of data inconsistency.

Data redundancy occurs when data of the same object is repeatedly defined multiple times within a single database. The redundancy creates space inefficiency and may produce data inconsistency.

File: 2022-OCT/answer.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+- Insertion Anomaly: Insertion of new student data requires inserting redundant club, activity and advisor data.
+- Modification Anomaly: Modification of the activity time of S011 is required to perform to all 2 records that have this activity to prevent data inconsistency.
+- Deletion Anomaly: Deletion of the club name 'Volleyball' will remove the student information S423 from the database.
+
+b)
+
+**1NF**
+
+CLUB(<ins>ClubCOde</ins>, ClubName, ActivityDay, ActivityTime, AdvisorID, AdvisorName, <ins>StudentID</ins>, StudentName, ContactNo, Programme)
```
Comment (YKLoh666) [inline 2022-OCT/answer.md]: <ins>ClubCOde</ins>

File: 2022-OCT/answer.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+	SELECT MemberNo
+	FROM Member
+	WHERE MemberName = 'Samatha Suzy'
+);
+```
+
+e)
+
+```sql
+SELECT p.PublisherName, COUNT(b.CallNum) books_published_2022
```
Comment (YKLoh666) [inline 2022-OCT/answer.md]: use alias for PublisherName

File: 2022-OCT/answer.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+FROM Book b
+JOIN Publisher p ON b.PublisherNo = p.PublisherNo
+WHERE YEAR(PublishedDate) = 2022
```
Comment (YKLoh666) [inline 2022-OCT/answer.md]: i don't think YEAR() working for oracle SQL.
use range instead 
```sql
WHERE PublishedDate BETWEEN '1-JAN-2022' AND '31-DEC-2022'
```

**Review by YKLoh666 — CHANGES_REQUESTED** [review]

File: 2022-OCT/answer.md (hunk 2 above)
Comment (zhijinggg) [reply 2022-OCT/answer.md:33]: 1 mark for the point, 1 mark for the elaboration ba i think enuf d?

**Review by YKLoh666 — APPROVED** [review]

**Pull request merged**